### ISSUE TITLE
Version 50 is causing Deadlocks when App DB is Postgres 

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#47660]
### Describe the bug

Metabase is hitting a number of deadlock issues when there are a lot of parameters on a chart. Something that didn't use to happen on 49. So some of the cards will not load due to a deadlock error, causing something like this:

<img width="1214" alt="image" src="https://github.com/user-attachments/assets/e0ef1aed-a5c8-466f-985b-ba459eba4bdd">

I believe this is causing other issues in other features like serialisation. Cause I am following another issue that has this error:

```
ERROR serialization.api :: Error during deserialization: Failed to load into database for Card RUvyr0HeZx4Si-k2SrN8Q {:path [{:id "RUvyr0HeZx4Si-k2SrN8Q", :model "Card"}], :deps-chain #{}, :model "Card", :table :report_card, :error :metabase-enterprise.serialization.v2.load/load-failure}
2024-09-04 08:55:58,254 ERROR serialization.api ::   caused by: ERROR: deadlock detected
  Detail: Process 1204 waits for ShareLock on transaction 2424836558; blocked by process 30145.
Process 30145 waits for ShareLock on transaction 2424835949; blocked by process 1204.
  Hint: See server log for query details.
  Where: while updating tuple (3,5) in relation "report_card" {}
2024-09-04 08:55:58,254 ERROR serialization.api ::   caused by: ERROR: deadlock detected
  Detail: Process 1204 waits for ShareLock on transaction 2424836558; blocked by process 30145.
Process 30145 waits for ShareLock on transaction 2424835949; blocked by process 1204.
  Hint: See server log for query details.
  Where: while updating tuple (3,5) in relation "report_card" 
2024-09-04 08:55:58,254 ERROR serialization.api :: Error during deserialization: Failed to load into database for Card RUvyr0HeZx4Si-k2SrN8Q {:path [{:id "RUvyr0HeZx4Si-k2SrN8Q", :model "Card"}], :deps-chain #{}, :model "Card", :table :report_card, :error :metabase-enterprise.serialization.v2.load/load-failure}
2024-09-04 08:55:58,254 ERROR serialization.api ::   caused by: ERROR: deadlock detected
2024-09-04 08:55:58,254 ERROR serialization.api ::   caused by: ERROR: deadlock detected
```

### To Reproduce

Make sure Metabase is running a Postges Application DB since I was not able to replicate this on H2

1. Go to New -> Question -> Product -> Save -> Add this Question 25 times into a dashbaord
2. Then add 3 filters and link them to Vendor, Ean, Title -> Finally select around 130 items from Title Filter:

<img width="1240" alt="image" src="https://github.com/user-attachments/assets/6b90c9bd-ad0e-48b8-8c9d-ad92a7eaf68e">

3. If you load this dashboard (inside metabase or embedded) you will notice some charts failing to load and an error in the logs as well

I also created a public link [on a cloud instance](https://metabasetest.metabaseapp.com/public/dashboard/2c2e01e6-7ac5-45e3-90ee-64f50b850db2) I have that will show up the error. Tried on stats but wasn't able to replicate

### Expected behavior

No deadlock errors or if a deadlock error happens metabase will retry to execute the query not fail the question

### Logs

```
[43d3414a-9423-43ff-a740-9be048810216] 2024-09-05T10:53:17+02:00 ERROR metabase.server.middleware.log POST /api/dashboard/199/dashcard/289/card/331/query 500 13.3 s (8 DB calls) {:metabase-user-id 1} 
{:via
 [{:type clojure.lang.ExceptionInfo,
   :message
   "ERROR: deadlock detected\n  Detail: Process 13367 waits for ShareLock on transaction 1806773214; blocked by process 13361.\nProcess 13361 waits for ShareLock on transaction 1806773197; blocked by process 13367.\n  Hint: See server log for query details.\n  Where: while locking tuple (2,3) in relation \"user_parameter_value\"",
   :data
   {:toucan2/context-trace
    [["execute SQL with class com.mchange.v2.c3p0.impl.NewProxyConnection"
      {:toucan2.jdbc.query/sql-args
       ["UPDATE \"user_parameter_value\" SET \"value\" = ? WHERE (\"user_id\" = ?) AND (\"dashboard_id\" = ?) AND (\"parameter_id\" = ?)"
        "[\"Sleek Aluminum Clock\",\"Mediocre Cotton Coat\",\"Durable Wool Toucan\",\"Lightweight Wool Plate\",\"Awesome Iron Hat\",\"Rustic Paper Bench\",\"Small Rubber Clock\",\"Synergistic Leather Coat\",\"Awesome Bronze Plate\",\"Aerodynamic Paper Coat\",\"Ergonomic Granite Bottle\",\"Lightweight Steel Knife\",\"Heavy-Duty Linen Toucan\",\"Durable Cotton Bench\",\"Aerodynamic Leather Computer\",\"Aerodynamic Granite Bottle\",\"Incredible Linen Knife\",\"Sleek Marble Table\",\"Mediocre Rubber Shoes\",\"Rustic Silk Knife\",\"Mediocre Paper Car\",\"Gorgeous Marble Plate\",\"Gorgeous Aluminum Plate\",\"Enormous Granite Wallet\",\"Enormous Cotton Pants\",\"Enormous Steel Watch\",\"Aerodynamic Leather Toucan\",\"Aerodynamic Concrete Bench\",\"Fantastic Aluminum Bottle\",\"Practical Copper Car\",\"Awesome Aluminum Table\",\"Rustic Iron Bench\",\"Small Granite Gloves\",\"Enormous Marble Gloves\",\"Enormous Plastic Coat\",\"Sleek Steel Table\",\"Ergonomic Marble Computer\",\"Enormous Granite Bottle\",\"Awesome Aluminum Keyboard\",\"Heavy-Duty Rubber Gloves\",\"Heavy-Duty Silk Car\",\"Sleek Wool Wallet\",\"Durable Leather Wallet\",\"Heavy-Duty Silk Chair\",\"Fantastic Copper Hat\",\"Practical Plastic Keyboard\",\"Durable Aluminum Bag\",\"Intelligent Paper Hat\",\"Rustic Rubber Clock\",\"Aerodynamic Granite Bench\",\"Practical Aluminum Coat\",\"Mediocre Marble Lamp\",\"Incredible Bronze Wallet\",\"Rustic Concrete Bottle\",\"Incredible Concrete Keyboard\",\"Awesome Cotton Shoes\",\"Durable Iron Knife\",\"Ergonomic Iron Watch\",\"Sleek Bronze Lamp\",\"Fantastic Steel Knife\",\"Ergonomic Marble Hat\",\"Rustic Paper Wallet\",\"Rustic Linen Keyboard\",\"Lightweight Steel Watch\",\"Sleek Aluminum Watch\",\"Practical Bronze Watch\",\"Lightweight Bronze Table\",\"Rustic Copper Hat\",\"Mediocre Wool Toucan\",\"Rustic Copper Knife\",\"Intelligent Bronze Knife\",\"Lightweight Paper Bottle\",\"Synergistic Rubber Shoes\",\"Lightweight Linen Hat\",\"Lightweight Linen Bottle\",\"Durable Steel Toucan\",\"Durable Rubber Computer\",\"Small Copper Plate\",\"Small Cotton Chair\",\"Intelligent Steel Car\",\"Durable Copper Clock\",\"Mediocre Aluminum Lamp\",\"Incredible Bronze Pants\",\"Gorgeous Concrete Chair\",\"Incredible Plastic Chair\",\"Ergonomic Paper Wallet\",\"Practical Silk Computer\",\"Practical Steel Table\",\"Sleek Wool Watch\",\"Lightweight Leather Gloves\",\"Sleek Plastic Shoes\",\"Enormous Marble Shoes\",\"Lightweight Granite Hat\",\"Sleek Marble Clock\",\"Small Plastic Computer\",\"Aerodynamic Concrete Lamp\",\"Intelligent Wooden Gloves\",\"Practical Aluminum Table\",\"Enormous Copper Shirt\",\"Synergistic Marble Keyboard\",\"Heavy-Duty Linen Gloves\",\"Small Marble Knife\",\"Mediocre Cotton Toucan\",\"Small Wool Wallet\",\"Synergistic Copper Computer\",\"Synergistic Granite Chair\",\"Synergistic Steel Chair\",\"Synergistic Wool Coat\",\"Small Marble Shoes\",\"Small Marble Hat\",\"Small Copper Clock\",\"Sleek Granite Pants\",\"Sleek Leather Table\",\"Sleek Leather Toucan\",\"Sleek Copper Watch\",\"Rustic Silk Pants\",\"Rustic Paper Car\",\"Rustic Rubber Knife\",\"Lightweight Copper Wallet\",\"Lightweight Leather Bench\",\"Lightweight Linen Coat\",\"Lightweight Marble Bag\",\"Intelligent Paper Car\",\"Intelligent Iron Shirt\",\"Incredible Silk Shoes\",\"Intelligent Granite Hat\",\"Practical Wool Hat\",\"Rustic Iron Keyboard\",\"Rustic Marble Bottle\",\"Sleek Paper Toucan\",\"Small Concrete Knife\"]"
        1
        199
        "e2f7a863"]}]
     ["resolve connection" {:toucan2.connection/connectable org.postgresql.jdbc.PgConnection}]
     ["resolve connection" {:toucan2.connection/connectable nil}]
     {:toucan2.pipeline/rf
      #object[clojure.core$completing$fn__8528 0x524f7e08 "clojure.core$completing$fn__8528@524f7e08"]}
     ["with compiled query"
      {:toucan2.pipeline/compiled-query
       ["UPDATE \"user_parameter_value\" SET \"value\" = ? WHERE (\"user_id\" = ?) AND (\"dashboard_id\" = ?) AND (\"parameter_id\" = ?)"
        "[\"Sleek Aluminum Clock\",\"Mediocre Cotton Coat\",\"Durable Wool Toucan\",\"Lightweight Wool Plate\",\"Awesome Iron Hat\",\"Rustic Paper Bench\",\"Small Rubber Clock\",\"Synergistic Leather Coat\",\"Awesome Bronze Plate\",\"Aerodynamic Paper Coat\",\"Ergonomic Granite Bottle\",\"Lightweight Steel Knife\",\"Heavy-Duty Linen Toucan\",\"Durable Cotton Bench\",\"Aerodynamic Leather Computer\",\"Aerodynamic Granite Bottle\",\"Incredible Linen Knife\",\"Sleek Marble Table\",\"Mediocre Rubber Shoes\",\"Rustic Silk Knife\",\"Mediocre Paper Car\",\"Gorgeous Marble Plate\",\"Gorgeous Aluminum Plate\",\"Enormous Granite Wallet\",\"Enormous Cotton Pants\",\"Enormous Steel Watch\",\"Aerodynamic Leather Toucan\",\"Aerodynamic Concrete Bench\",\"Fantastic Aluminum Bottle\",\"Practical Copper Car\",\"Awesome Aluminum Table\",\"Rustic Iron Bench\",\"Small Granite Gloves\",\"Enormous Marble Gloves\",\"Enormous Plastic Coat\",\"Sleek Steel Table\",\"Ergonomic Marble Computer\",\"Enormous Granite Bottle\",\"Awesome Aluminum Keyboard\",\"Heavy-Duty Rubber Gloves\",\"Heavy-Duty Silk Car\",\"Sleek Wool Wallet\",\"Durable Leather Wallet\",\"Heavy-Duty Silk Chair\",\"Fantastic Copper Hat\",\"Practical Plastic Keyboard\",\"Durable Aluminum Bag\",\"Intelligent Paper Hat\",\"Rustic Rubber Clock\",\"Aerodynamic Granite Bench\",\"Practical Aluminum Coat\",\"Mediocre Marble Lamp\",\"Incredible Bronze Wallet\",\"Rustic Concrete Bottle\",\"Incredible Concrete Keyboard\",\"Awesome Cotton Shoes\",\"Durable Iron Knife\",\"Ergonomic Iron Watch\",\"Sleek Bronze Lamp\",\"Fantastic Steel Knife\",\"Ergonomic Marble Hat\",\"Rustic Paper Wallet\",\"Rustic Linen Keyboard\",\"Lightweight Steel Watch\",\"Sleek Aluminum Watch\",\"Practical Bronze Watch\",\"Lightweight Bronze Table\",\"Rustic Copper Hat\",\"Mediocre Wool Toucan\",\"Rustic Copper Knife\",\"Intelligent Bronze Knife\",\"Lightweight Paper Bottle\",\"Synergistic Rubber Shoes\",\"Lightweight Linen Hat\",\"Lightweight Linen Bottle\",\"Durable Steel Toucan\",\"Durable Rubber Computer\",\"Small Copper Plate\",\"Small Cotton Chair\",\"Intelligent Steel Car\",\"Durable Copper Clock\",\"Mediocre Aluminum Lamp\",\"Incredible Bronze Pants\",\"Gorgeous Concrete Chair\",\"Incredible Plastic Chair\",\"Ergonomic Paper Wallet\",\"Practical Silk Computer\",\"Practical Steel Table\",\"Sleek Wool Watch\",\"Lightweight Leather Gloves\",\"Sleek Plastic Shoes\",\"Enormous Marble Shoes\",\"Lightweight Granite Hat\",\"Sleek Marble Clock\",\"Small Plastic Computer\",\"Aerodynamic Concrete Lamp\",\"Intelligent Wooden Gloves\",\"Practical Aluminum Table\",\"Enormous Copper Shirt\",\"Synergistic Marble Keyboard\",\"Heavy-Duty Linen Gloves\",\"Small Marble Knife\",\"Mediocre Cotton Toucan\",\"Small Wool Wallet\",\"Synergistic Copper Computer\",\"Synergistic Granite Chair\",\"Synergistic Steel Chair\",\"Synergistic Wool Coat\",\"Small Marble Shoes\",\"Small Marble Hat\",\"Small Copper Clock\",\"Sleek Granite Pants\",\"Sleek Leather Table\",\"Sleek Leather Toucan\",\"Sleek Copper Watch\",\"Rustic Silk Pants\",\"Rustic Paper Car\",\"Rustic Rubber Knife\",\"Lightweight Copper Wallet\",\"Lightweight Leather Bench\",\"Lightweight Linen Coat\",\"Lightweight Marble Bag\",\"Intelligent Paper Car\",\"Intelligent Iron Shirt\",\"Incredible Silk Shoes\",\"Intelligent Granite Hat\",\"Practical Wool Hat\",\"Rustic Iron Keyboard\",\"Rustic Marble Bottle\",\"Sleek Paper Toucan\",\"Small Concrete Knife\"]"
        1
        199
        "e2f7a863"]}]
     ["with built query"
      {:toucan2.pipeline/built-query
       {:update [:user_parameter_value],
        :set
        {:value
         "[\"Sleek Aluminum Clock\",\"Mediocre Cotton Coat\",\"Durable Wool Toucan\",\"Lightweight Wool Plate\",\"Awesome Iron Hat\",\"Rustic Paper Bench\",\"Small Rubber Clock\",\"Synergistic Leather Coat\",\"Awesome Bronze Plate\",\"Aerodynamic Paper Coat\",\"Ergonomic Granite Bottle\",\"Lightweight Steel Knife\",\"Heavy-Duty Linen Toucan\",\"Durable Cotton Bench\",\"Aerodynamic Leather Computer\",\"Aerodynamic Granite Bottle\",\"Incredible Linen Knife\",\"Sleek Marble Table\",\"Mediocre Rubber Shoes\",\"Rustic Silk Knife\",\"Mediocre Paper Car\",\"Gorgeous Marble Plate\",\"Gorgeous Aluminum Plate\",\"Enormous Granite Wallet\",\"Enormous Cotton Pants\",\"Enormous Steel Watch\",\"Aerodynamic Leather Toucan\",\"Aerodynamic Concrete Bench\",\"Fantastic Aluminum Bottle\",\"Practical Copper Car\",\"Awesome Aluminum Table\",\"Rustic Iron Bench\",\"Small Granite Gloves\",\"Enormous Marble Gloves\",\"Enormous Plastic Coat\",\"Sleek Steel Table\",\"Ergonomic Marble Computer\",\"Enormous Granite Bottle\",\"Awesome Aluminum Keyboard\",\"Heavy-Duty Rubber Gloves\",\"Heavy-Duty Silk Car\",\"Sleek Wool Wallet\",\"Durable Leather Wallet\",\"Heavy-Duty Silk Chair\",\"Fantastic Copper Hat\",\"Practical Plastic Keyboard\",\"Durable Aluminum Bag\",\"Intelligent Paper Hat\",\"Rustic Rubber Clock\",\"Aerodynamic Granite Bench\",\"Practical Aluminum Coat\",\"Mediocre Marble Lamp\",\"Incredible Bronze Wallet\",\"Rustic Concrete Bottle\",\"Incredible Concrete Keyboard\",\"Awesome Cotton Shoes\",\"Durable Iron Knife\",\"Ergonomic Iron Watch\",\"Sleek Bronze Lamp\",\"Fantastic Steel Knife\",\"Ergonomic Marble Hat\",\"Rustic Paper Wallet\",\"Rustic Linen Keyboard\",\"Lightweight Steel Watch\",\"Sleek Aluminum Watch\",\"Practical Bronze Watch\",\"Lightweight Bronze Table\",\"Rustic Copper Hat\",\"Mediocre Wool Toucan\",\"Rustic Copper Knife\",\"Intelligent Bronze Knife\",\"Lightweight Paper Bottle\",\"Synergistic Rubber Shoes\",\"Lightweight Linen Hat\",\"Lightweight Linen Bottle\",\"Durable Steel Toucan\",\"Durable Rubber Computer\",\"Small Copper Plate\",\"Small Cotton Chair\",\"Intelligent Steel Car\",\"Durable Copper Clock\",\"Mediocre Aluminum Lamp\",\"Incredible Bronze Pants\",\"Gorgeous Concrete Chair\",\"Incredible Plastic Chair\",\"Ergonomic Paper Wallet\",\"Practical Silk Computer\",\"Practical Steel Table\",\"Sleek Wool Watch\",\"Lightweight Leather Gloves\",\"Sleek Plastic Shoes\",\"Enormous Marble Shoes\",\"Lightweight Granite Hat\",\"Sleek Marble Clock\",\"Small Plastic Computer\",\"Aerodynamic Concrete Lamp\",\"Intelligent Wooden Gloves\",\"Practical Aluminum Table\",\"Enormous Copper Shirt\",\"Synergistic Marble Keyboard\",\"Heavy-Duty Linen Gloves\",\"Small Marble Knife\",\"Mediocre Cotton Toucan\",\"Small Wool Wallet\",\"Synergistic Copper Computer\",\"Synergistic Granite Chair\",\"Synergistic Steel Chair\",\"Synergistic Wool Coat\",\"Small Marble Shoes\",\"Small Marble Hat\",\"Small Copper Clock\",\"Sleek Granite Pants\",\"Sleek Leather Table\",\"Sleek Leather Toucan\",\"Sleek Copper Watch\",\"Rustic Silk Pants\",\"Rustic Paper Car\",\"Rustic Rubber Knife\",\"Lightweight Copper Wallet\",\"Lightweight Leather Bench\",\"Lightweight Linen Coat\",\"Lightweight Marble Bag\",\"Intelligent Paper Car\",\"Intelligent Iron Shirt\",\"Incredible Silk Shoes\",\"Intelligent Granite Hat\",\"Practical Wool Hat\",\"Rustic Iron Keyboard\",\"Rustic Marble Bottle\",\"Sleek Paper Toucan\",\"Small Concrete Knife\"]"},
        :where [:and [:= :user_id 1] [:= :dashboard_id 199] [:= :parameter_id "e2f7a863"]]}}]
     ["with resolved query"
      {:toucan2.pipeline/resolved-query {:user_id 1, :dashboard_id 199, :parameter_id "e2f7a863"}}]
     ["with parsed args"
      {:toucan2.pipeline/query-type :toucan.query-type/update.update-count,
       :toucan2.pipeline/parsed-args
       {:queryable {:user_id 1, :dashboard_id 199, :parameter_id "e2f7a863"},
        :changes
        {:value
         ["Sleek Aluminum Clock"
          "Mediocre Cotton Coat"
          "Durable Wool Toucan"
          "Lightweight Wool Plate"
          "Awesome Iron Hat"
          "Rustic Paper Bench"
          "Small Rubber Clock"
          "Synergistic Leather Coat"
          "Awesome Bronze Plate"
          "Aerodynamic Paper Coat"
          "Ergonomic Granite Bottle"
          "Lightweight Steel Knife"
          "Heavy-Duty Linen Toucan"
          "Durable Cotton Bench"
          "Aerodynamic Leather Computer"
          "Aerodynamic Granite Bottle"
          "Incredible Linen Knife"
          "Sleek Marble Table"
          "Mediocre Rubber Shoes"
          "Rustic Silk Knife"
          "Mediocre Paper Car"
          "Gorgeous Marble Plate"
          "Gorgeous Aluminum Plate"
          "Enormous Granite Wallet"
          "Enormous Cotton Pants"
          "Enormous Steel Watch"
          "Aerodynamic Leather Toucan"
          "Aerodynamic Concrete Bench"
          "Fantastic Aluminum Bottle"
          "Practical Copper Car"
          "Awesome Aluminum Table"
          "Rustic Iron Bench"
          "Small Granite Gloves"
          "Enormous Marble Gloves"
          "Enormous Plastic Coat"
          "Sleek Steel Table"
          "Ergonomic Marble Computer"
          "Enormous Granite Bottle"
          "Awesome Aluminum Keyboard"
          "Heavy-Duty Rubber Gloves"
          "Heavy-Duty Silk Car"
          "Sleek Wool Wallet"
          "Durable Leather Wallet"
          "Heavy-Duty Silk Chair"
          "Fantastic Copper Hat"
          "Practical Plastic Keyboard"
          "Durable Aluminum Bag"
          "Intelligent Paper Hat"
          "Rustic Rubber Clock"
          "Aerodynamic Granite Bench"
          "Practical Aluminum Coat"
          "Mediocre Marble Lamp"
          "Incredible Bronze Wallet"
          "Rustic Concrete Bottle"
          "Incredible Concrete Keyboard"
          "Awesome Cotton Shoes"
          "Durable Iron Knife"
          "Ergonomic Iron Watch"
          "Sleek Bronze Lamp"
          "Fantastic Steel Knife"
          "Ergonomic Marble Hat"
          "Rustic Paper Wallet"
          "Rustic Linen Keyboard"
          "Lightweight Steel Watch"
          "Sleek Aluminum Watch"
          "Practical Bronze Watch"
          "Lightweight Bronze Table"
          "Rustic Copper Hat"
          "Mediocre Wool Toucan"
          "Rustic Copper Knife"
          "Intelligent Bronze Knife"
          "Lightweight Paper Bottle"
          "Synergistic Rubber Shoes"
          "Lightweight Linen Hat"
          "Lightweight Linen Bottle"
          "Durable Steel Toucan"
          "Durable Rubber Computer"
          "Small Copper Plate"
          "Small Cotton Chair"
          "Intelligent Steel Car"
          "Durable Copper Clock"
          "Mediocre Aluminum Lamp"
          "Incredible Bronze Pants"
          "Gorgeous Concrete Chair"
          "Incredible Plastic Chair"
          "Ergonomic Paper Wallet"
          "Practical Silk Computer"
          "Practical Steel Table"
          "Sleek Wool Watch"
          "Lightweight Leather Gloves"
          "Sleek Plastic Shoes"
          "Enormous Marble Shoes"
          "Lightweight Granite Hat"
          "Sleek Marble Clock"
          "Small Plastic Computer"
          "Aerodynamic Concrete Lamp"
          "Intelligent Wooden Gloves"
          "Practical Aluminum Table"
          "Enormous Copper Shirt"
          "Synergistic Marble Keyboard"
          "Heavy-Duty Linen Gloves"
          "Small Marble Knife"
          "Mediocre Cotton Toucan"
          "Small Wool Wallet"
          "Synergistic Copper Computer"
          "Synergistic Granite Chair"
          "Synergistic Steel Chair"
          "Synergistic Wool Coat"
          "Small Marble Shoes"
          "Small Marble Hat"
          "Small Copper Clock"
          "Sleek Granite Pants"
          "Sleek Leather Table"
          "Sleek Leather Toucan"
          "Sleek Copper Watch"
          "Rustic Silk Pants"
          "Rustic Paper Car"
          "Rustic Rubber Knife"
          "Lightweight Copper Wallet"
          "Lightweight Leather Bench"
          "Lightweight Linen Coat"
          "Lightweight Marble Bag"
          "Intelligent Paper Car"
          "Intelligent Iron Shirt"
          "Incredible Silk Shoes"
          "Intelligent Granite Hat"
          "Practical Wool Hat"
          "Rustic Iron Keyboard"
          "Rustic Marble Bottle"
          "Sleek Paper Toucan"
          "Small Concrete Knife"]}}}]
     ["with model" {:toucan2.pipeline/model :model/UserParameterValue}]
     ["with unparsed args"
      {:toucan2.pipeline/query-type :toucan.query-type/update.update-count,
       :toucan2.pipeline/unparsed-args
       (:model/UserParameterValue
        {:user_id 1, :dashboard_id 199, :parameter_id "e2f7a863"}
        {:value
         ["Sleek Aluminum Clock"
          "Mediocre Cotton Coat"
          "Durable Wool Toucan"
          "Lightweight Wool Plate"
          "Awesome Iron Hat"
          "Rustic Paper Bench"
          "Small Rubber Clock"
          "Synergistic Leather Coat"
          "Awesome Bronze Plate"
          "Aerodynamic Paper Coat"
          "Ergonomic Granite Bottle"
          "Lightweight Steel Knife"
          "Heavy-Duty Linen Toucan"
          "Durable Cotton Bench"
          "Aerodynamic Leather Computer"
          "Aerodynamic Granite Bottle"
          "Incredible Linen Knife"
          "Sleek Marble Table"
          "Mediocre Rubber Shoes"
          "Rustic Silk Knife"
          "Mediocre Paper Car"
          "Gorgeous Marble Plate"
          "Gorgeous Aluminum Plate"
          "Enormous Granite Wallet"
          "Enormous Cotton Pants"
          "Enormous Steel Watch"
          "Aerodynamic Leather Toucan"
          "Aerodynamic Concrete Bench"
          "Fantastic Aluminum Bottle"
          "Practical Copper Car"
          "Awesome Aluminum Table"
          "Rustic Iron Bench"
          "Small Granite Gloves"
          "Enormous Marble Gloves"
          "Enormous Plastic Coat"
          "Sleek Steel Table"
          "Ergonomic Marble Computer"
          "Enormous Granite Bottle"
          "Awesome Aluminum Keyboard"
          "Heavy-Duty Rubber Gloves"
          "Heavy-Duty Silk Car"
          "Sleek Wool Wallet"
          "Durable Leather Wallet"
          "Heavy-Duty Silk Chair"
          "Fantastic Copper Hat"
          "Practical Plastic Keyboard"
          "Durable Aluminum Bag"
          "Intelligent Paper Hat"
          "Rustic Rubber Clock"
          "Aerodynamic Granite Bench"
          "Practical Aluminum Coat"
          "Mediocre Marble Lamp"
          "Incredible Bronze Wallet"
          "Rustic Concrete Bottle"
          "Incredible Concrete Keyboard"
          "Awesome Cotton Shoes"
          "Durable Iron Knife"
          "Ergonomic Iron Watch"
          "Sleek Bronze Lamp"
          "Fantastic Steel Knife"
          "Ergonomic Marble Hat"
          "Rustic Paper Wallet"
          "Rustic Linen Keyboard"
          "Lightweight Steel Watch"
          "Sleek Aluminum Watch"
          "Practical Bronze Watch"
          "Lightweight Bronze Table"
          "Rustic Copper Hat"
          "Mediocre Wool Toucan"
          "Rustic Copper Knife"
          "Intelligent Bronze Knife"
          "Lightweight Paper Bottle"
          "Synergistic Rubber Shoes"
          "Lightweight Linen Hat"
          "Lightweight Linen Bottle"
          "Durable Steel Toucan"
          "Durable Rubber Computer"
          "Small Copper Plate"
          "Small Cotton Chair"
          "Intelligent Steel Car"
          "Durable Copper Clock"
          "Mediocre Aluminum Lamp"
          "Incredible Bronze Pants"
          "Gorgeous Concrete Chair"
          "Incredible Plastic Chair"
          "Ergonomic Paper Wallet"
          "Practical Silk Computer"
          "Practical Steel Table"
          "Sleek Wool Watch"
          "Lightweight Leather Gloves"
          "Sleek Plastic Shoes"
          "Enormous Marble Shoes"
          "Lightweight Granite Hat"
          "Sleek Marble Clock"
          "Small Plastic Computer"
          "Aerodynamic Concrete Lamp"
          "Intelligent Wooden Gloves"
          "Practical Aluminum Table"
          "Enormous Copper Shirt"
          "Synergistic Marble Keyboard"
          "Heavy-Duty Linen Gloves"
          "Small Marble Knife"
          "Mediocre Cotton Toucan"
          "Small Wool Wallet"
          "Synergistic Copper Computer"
          "Synergistic Granite Chair"
          "Synergistic Steel Chair"
          "Synergistic Wool Coat"
          "Small Marble Shoes"
          "Small Marble Hat"
          "Small Copper Clock"
          "Sleek Granite Pants"
          "Sleek Leather Table"
          "Sleek Leather Toucan"
          "Sleek Copper Watch"
          "Rustic Silk Pants"
          "Rustic Paper Car"
          "Rustic Rubber Knife"
          "Lightweight Copper Wallet"
          "Lightweight Leather Bench"
          "Lightweight Linen Coat"
          "Lightweight Marble Bag"
          "Intelligent Paper Car"
          "Intelligent Iron Shirt"
          "Incredible Silk Shoes"
          "Intelligent Granite Hat"
          "Practical Wool Hat"
          "Rustic Iron Keyboard"
          "Rustic Marble Bottle"
          "Sleek Paper Toucan"
          "Small Concrete Knife"]})}]
     ["resolve connection" {:toucan2.connection/connectable metabase.db.connection.ApplicationDB}]
     ["resolve connection" {:toucan2.connection/connectable :default}]
     ["resolve connection" {:toucan2.connection/connectable nil}]]},
   :at [org.postgresql.core.v3.QueryExecutorImpl receiveErrorResponse "QueryExecutorImpl.java" 2725]}
  {:type org.postgresql.util.PSQLException,
   :message
   "ERROR: deadlock detected\n  Detail: Process 13367 waits for ShareLock on transaction 1806773214; blocked by process 13361.\nProcess 13361 waits for ShareLock on transaction 1806773197; blocked by process 13367.\n  Hint: See server log for query details.\n  Where: while locking tuple (2,3) in relation \"user_parameter_value\"",
   :at [org.postgresql.core.v3.QueryExecutorImpl receiveErrorResponse "QueryExecutorImpl.java" 2725]}],
 :trace
 [[org.postgresql.core.v3.QueryExecutorImpl receiveErrorResponse "QueryExecutorImpl.java" 2725]
  [org.postgresql.core.v3.QueryExecutorImpl processResults "QueryExecutorImpl.java" 2412]
  [org.postgresql.core.v3.QueryExecutorImpl execute "QueryExecutorImpl.java" 371]
  [org.postgresql.jdbc.PgStatement executeInternal "PgStatement.java" 502]
  [org.postgresql.jdbc.PgStatement execute "PgStatement.java" 419]
  [org.postgresql.jdbc.PgPreparedStatement executeWithFlags "PgPreparedStatement.java" 194]
  [org.postgresql.jdbc.PgPreparedStatement execute "PgPreparedStatement.java" 180]
  [com.mchange.v2.c3p0.impl.NewProxyPreparedStatement execute "NewProxyPreparedStatement.java" 67]
  [toucan2.jdbc.query$reduce_jdbc_query invokeStatic "query.clj" 40]
  [toucan2.jdbc.query$reduce_jdbc_query invoke "query.clj" 22]
  [toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default
   invokeStatic
   "pipeline.clj"
   19]
  [toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default
   invoke
   "pipeline.clj"
   9]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 15]
  [methodical.impl.combo.threaded$combine_methods_thread_last$fn__18430$combined_method_thread_last__18431
   invoke
   "threaded.clj"
   64]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 65]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 216]
  [toucan2.pipeline$transduce_execute$with_connection_STAR___21534$with_transaction_STAR___21535
   invoke
   "pipeline.clj"
   75]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invokeStatic "connection.clj" 180]
  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invoke "connection.clj" 162]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 13]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invokeStatic "connection.clj" 249]
  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invoke "connection.clj" 245]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 13]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 58]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 212]
  [toucan2.pipeline$transduce_execute$with_connection_STAR___21534 invoke "pipeline.clj" 74]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection invokeStatic "connection.clj" 13]
  [toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection invoke "connection.clj" 11]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic "connection.clj" 118]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke "connection.clj" 106]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 55]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 210]
  [toucan2.connection$do_with_connection_primary_method_ invokeStatic "connection.clj" 204]
  [toucan2.connection$do_with_connection_primary_method_ invoke "connection.clj" 194]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic "connection.clj" 118]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke "connection.clj" 106]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 55]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 210]
  [toucan2.pipeline$transduce_execute invokeStatic "pipeline.clj" 74]
  [toucan2.pipeline$transduce_execute invoke "pipeline.clj" 64]
  [clojure.lang.Var invoke "Var.java" 399]
  [toucan2.pipeline$transduce_compiled_query invokeStatic "pipeline.clj" 244]
  [toucan2.pipeline$transduce_compiled_query invoke "pipeline.clj" 240]
  [toucan2.pipeline$transduce_built_query invokeStatic "pipeline.clj" 252]
  [toucan2.pipeline$transduce_built_query invoke "pipeline.clj" 246]
  [toucan2.pipeline$transduce_query_primary_method_default invokeStatic "pipeline.clj" 272]
  [toucan2.pipeline$transduce_query_primary_method_default invoke "pipeline.clj" 269]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 15]
  [methodical.impl.combo.threaded$combine_methods_thread_last$fn__18430$combined_method_thread_last__18431
   invoke
   "threaded.clj"
   64]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 65]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 216]
  [toucan2.pipeline$transduce_query_STAR_ invokeStatic "pipeline.clj" 278]
  [toucan2.pipeline$transduce_query_STAR_ invoke "pipeline.clj" 274]
  [toucan2.pipeline$transduce_with_model invokeStatic "pipeline.clj" 293]
  [toucan2.pipeline$transduce_with_model invoke "pipeline.clj" 280]
  [toucan2.pipeline$transduce_parsed invokeStatic "pipeline.clj" 309]
  [toucan2.pipeline$transduce_parsed invoke "pipeline.clj" 295]
  [toucan2.pipeline$transduce_unparsed invokeStatic "pipeline.clj" 317]
  [toucan2.pipeline$transduce_unparsed invoke "pipeline.clj" 311]
  [toucan2.pipeline$transduce_unparsed_with_default_rf invokeStatic "pipeline.clj" 374]
  [toucan2.pipeline$transduce_unparsed_with_default_rf invoke "pipeline.clj" 368]
  [toucan2.update$update_BANG_ invokeStatic "update.clj" 53]
  [toucan2.update$update_BANG_ doInvoke "update.clj" 50]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [metabase.models.user_parameter_value$batched_upsert_BANG_$with_connection_STAR___98119$with_transaction_STAR___98120
   invoke
   "user_parameter_value.clj"
   45]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [metabase.db.connection$do_transaction$thunk__42360 invoke "connection.clj" 140]
  [metabase.db.connection$do_transaction invokeStatic "connection.clj" 152]
  [metabase.db.connection$do_transaction invoke "connection.clj" 136]
  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invokeStatic "connection.clj" 189]
  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invoke "connection.clj" 162]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 13]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invokeStatic "connection.clj" 249]
  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invoke "connection.clj" 245]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 13]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 58]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 212]
  [metabase.models.user_parameter_value$batched_upsert_BANG_$with_connection_STAR___98119
   invoke
   "user_parameter_value.clj"
   43]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [toucan2.connection$bind_current_connectable_fn$fn__21215 invoke "connection.clj" 104]
  [toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource invokeStatic "connection.clj" 18]
  [toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource invoke "connection.clj" 15]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic "connection.clj" 118]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke "connection.clj" 106]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 55]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 210]
  [metabase.db.connection$do_with_connection_primary_method_default invokeStatic "connection.clj" 132]
  [metabase.db.connection$do_with_connection_primary_method_default invoke "connection.clj" 130]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic "connection.clj" 118]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke "connection.clj" 106]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 55]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 210]
  [toucan2.connection$do_with_connection_primary_method_ invokeStatic "connection.clj" 204]
  [toucan2.connection$do_with_connection_primary_method_ invoke "connection.clj" 194]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic "connection.clj" 118]
  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke "connection.clj" 106]
  [methodical.impl.combo.common$partial_STAR_$fn__18120 invoke "common.clj" 12]
  [methodical.util.FnWithMeta invoke "util.clj" 46]
  [methodical.impl.standard$invoke_multifn invokeStatic "standard.clj" 55]
  [methodical.impl.standard$invoke_multifn invoke "standard.clj" 47]
  [methodical.impl.standard.StandardMultiFn invoke "standard.clj" 210]
  [metabase.models.user_parameter_value$batched_upsert_BANG_ invokeStatic "user_parameter_value.clj" 43]
  [metabase.models.user_parameter_value$batched_upsert_BANG_ invoke "user_parameter_value.clj" 33]
  [metabase.query_processor.dashboard$resolve_params_for_query invokeStatic "dashboard.clj" 147]
  [metabase.query_processor.dashboard$resolve_params_for_query invoke "dashboard.clj" 118]
  [metabase.query_processor.dashboard$process_query_for_dashcard invokeStatic "dashboard.clj" 182]
  [metabase.query_processor.dashboard$process_query_for_dashcard doInvoke "dashboard.clj" 163]
  [clojure.lang.RestFn applyTo "RestFn.java" 137]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$apply invoke "core.clj" 662]
  [medley.core$mapply invokeStatic "core.cljc" 243]
  [medley.core$mapply invoke "core.cljc" 237]
  [metabase.api.dashboard$fn__99040$fn__99043 invoke "dashboard.clj" 1240]
  [metabase.api.dashboard$do_with_dashboard_load_id invokeStatic "dashboard.clj" 307]
  [metabase.api.dashboard$do_with_dashboard_load_id invoke "dashboard.clj" 302]
  [metabase.api.dashboard$fn__99040 invokeStatic "dashboard.clj" 1239]
  [metabase.api.dashboard$fn__99040 invoke "dashboard.clj" 1231]
  [compojure.core$wrap_response$fn__52704 invoke "core.clj" 160]
  [compojure.core$wrap_route_middleware$fn__52688 invoke "core.clj" 132]
  [compojure.core$wrap_route_info$fn__52693 invoke "core.clj" 139]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 151]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 152]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 152]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 152]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 152]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [metabase.server.middleware.auth$enforce_authentication$fn__98049 invoke "auth.clj" 18]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$make_context$handler__52744 invoke "core.clj" 290]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 300]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [clojure.lang.Var invoke "Var.java" 393]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [metabase.api.routes$fn__105642$fn__105643 invoke "routes.clj" 70]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.lang.AFunction$1 doInvoke "AFunction.java" 31]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$apply invoke "core.clj" 662]
  [metabase.server.routes$fn__105922$fn__105923 doInvoke "routes.clj" 73]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$make_context$handler__52744 invoke "core.clj" 290]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 300]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__52697 invoke "core.clj" 153]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 199]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$make_context$handler__52744 invoke "core.clj" 290]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 300]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716$f__52717$respond_SINGLEQUOTE___52718 invoke "core.clj" 197]
  [compojure.core$make_context$fn__52748 invoke "core.clj" 301]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [compojure.core$routes$fn__52716$f__52717 invoke "core.clj" 198]
  [compojure.core$routes$fn__52716 invoke "core.clj" 200]
  [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__102022 invoke "exceptions.clj" 107]
  [metabase.server.middleware.exceptions$catch_api_exceptions$fn__102019 invoke "exceptions.clj" 96]
  [metabase.server.middleware.log$log_api_call$fn__108278$fn__108279$fn__108280 invoke "log.clj" 233]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic "diagnostic.clj" 18]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke "diagnostic.clj" 12]
  [metabase.server.middleware.log$log_api_call$fn__108278$fn__108279 invoke "log.clj" 224]
  [toucan2.execute$do_with_call_counts invokeStatic "execute.clj" 112]
  [toucan2.execute$do_with_call_counts invoke "execute.clj" 103]
  [metabase.server.middleware.log$log_api_call$fn__108278 invoke "log.clj" 223]
  [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__114444 invoke "browser_cookie.clj" 40]
  [metabase.server.middleware.security$add_security_headers$fn__101978 invoke "security.clj" 240]
  [ring.middleware.json$wrap_json_body$fn__114703 invoke "json.clj" 64]
  [metabase.server.middleware.offset_paging$handle_paging$fn__88221 invoke "offset_paging.clj" 43]
  [metabase.server.middleware.json$wrap_streamed_json_response$fn__54325 invoke "json.clj" 83]
  [ring.middleware.keyword_params$wrap_keyword_params$fn__114792 invoke "keyword_params.clj" 55]
  [ring.middleware.params$wrap_params$fn__114811 invoke "params.clj" 77]
  [metabase.server.middleware.misc$maybe_set_site_url$fn__70881 invoke "misc.clj" 59]
  [metabase.server.middleware.session$reset_session_timeout$fn__77889 invoke "session.clj" 548]
  [metabase.server.middleware.session$bind_current_user$fn__77855$fn__77856 invoke "session.clj" 443]
  [metabase.server.middleware.session$do_with_current_user invokeStatic "session.clj" 422]
  [metabase.server.middleware.session$do_with_current_user invoke "session.clj" 405]
  [metabase.server.middleware.session$bind_current_user$fn__77855 invoke "session.clj" 442]
  [metabase.server.middleware.session$wrap_current_user_info$fn__77836 invoke "session.clj" 381]
  [metabase.server.middleware.session$wrap_session_id$fn__77808 invoke "session.clj" 259]
  [metabase.server.middleware.auth$wrap_static_api_key$fn__98057 invoke "auth.clj" 32]
  [ring.middleware.cookies$wrap_cookies$fn__114631 invoke "cookies.clj" 200]
  [metabase.server.middleware.misc$add_content_type$fn__70863 invoke "misc.clj" 28]
  [metabase.server.middleware.misc$disable_streaming_buffering$fn__70889 invoke "misc.clj" 75]
  [ring.middleware.gzip$wrap_gzip$fn__114673 invoke "gzip.clj" 86]
  [metabase.server.middleware.misc$bind_request$fn__70892 invoke "misc.clj" 91]
  [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__114460 invoke "ssl.clj" 51]
  [metabase.server$async_proxy_handler$fn__71227 invoke "server.clj" 77]
  [metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a handle nil -1]
  [org.eclipse.jetty.server.handler.StatisticsHandler handle "StatisticsHandler.java" 173]
  [org.eclipse.jetty.server.handler.HandlerWrapper handle "HandlerWrapper.java" 122]
  [org.eclipse.jetty.server.Server handle "Server.java" 563]
  [org.eclipse.jetty.server.HttpChannel$RequestDispatchable dispatch "HttpChannel.java" 1598]
  [org.eclipse.jetty.server.HttpChannel dispatch "HttpChannel.java" 753]
  [org.eclipse.jetty.server.HttpChannel handle "HttpChannel.java" 501]
  [org.eclipse.jetty.server.HttpConnection onFillable "HttpConnection.java" 287]
  [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded "AbstractConnection.java" 314]
  [org.eclipse.jetty.io.FillInterest fillable "FillInterest.java" 100]
  [org.eclipse.jetty.io.SelectableChannelEndPoint$1 run "SelectableChannelEndPoint.java" 53]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy runTask "AdaptiveExecutionStrategy.java" 421]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy consumeTask "AdaptiveExecutionStrategy.java" 390]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy tryProduce "AdaptiveExecutionStrategy.java" 277]
  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy run "AdaptiveExecutionStrategy.java" 199]
  [org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread run "ReservedThreadExecutor.java" 411]
  [org.eclipse.jetty.util.thread.QueuedThreadPool runJob "QueuedThreadPool.java" 969]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner doRunJob "QueuedThreadPool.java" 1194]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run "QueuedThreadPool.java" 1149]
  [java.lang.Thread run nil -1]],
 :cause
 "ERROR: deadlock detected\n  Detail: Process 13367 waits for ShareLock on transaction 1806773214; blocked by process 13361.\nProcess 13361 waits for ShareLock on transaction 1806773197; blocked by process 13367.\n  Hint: See server log for query details.\n  Where: while locking tuple (2,3) in relation \"user_parameter_value\"",
 :message
 "ERROR: deadlock detected\n  Detail: Process 13367 waits for ShareLock on transaction 1806773214; blocked by process 13361.\nProcess 13361 waits for ShareLock on transaction 1806773197; blocked by process 13367.\n  Hint: See server log for query details.\n  Where: while locking tuple (2,3) in relation \"user_parameter_value\""}
```

### Information about your Metabase installation

```JSON
Was able to replicate on 50.24 ... Not able to hit on stats
```

### Severity

Broke dashboards of some customers

### Additional context

A workaround is to increase the App DB connection pool ... but this is not really a workaround but more of a hiding of the problem

**Comment by Tony-metabase** [comment]
It could be the fix here https://github.com/metabase/metabase/pull/46615 improved this some time ago but the issue is still there

**Comment by eli-noogata** [comment]
@paoliniluis hey, is there a new version that includes the fix?

**Comment by paoliniluis** [comment]
@eli-noogata, you might want to try 50.25.3 which should do the trick. Please try that and report back

**Comment by paoliniluis** [comment]
@eli-noogata can you check 52.5?
